go-runtime/compile/parser.go

```diff
@@ -1,5 +1,8 @@
 package compile
 
+// TODO: This file is now duplicated in go-runtime/schema/analyzers/parser.go. It should be removed
+// from here once the schema extraction refactoring is complete.
```

Comment:
just duplicated this file to avoid having to export everything from it's new location (so the legacy extractor can continue using it) only to make it all private again once the refactor is complete